### PR TITLE
chore(ui): Update types devDependencies in ui

### PR DESCRIPTION
**Pull request by pedrottimark**
## Description

Update sooner than later to reduce complexity of supporting releases for longer time period.

Especially updates to package types affect current code slightly, raise the lint standard for future code.

## Checklist
- [x] Investigated and inspected CI test results
- [x] ~Unit test and regression tests added~

## Testing Performed

1. `yarn lint` in ui/apps/platform
2. `yarn tsc` in ui/apps/platform because `yarn build` displays only the first error

    * https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/ExceptionConfiguration/VulnerabilitiesConfiguration.tsx#L289

        > TS2322: Type 'undefined' is not assignable to type 'string'.

        > The expected type comes from property 'helperTextInvalid' which is declared here on type 'IntrinsicAttributes & BaseSettingProps & { value: number; validated: "error" | "default" | "warning" | "success" | "indeterminate" | undefined; helperTextInvalid: string; }'

        Solution: Add `?? ''` nullish coalescing.

    * https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationContent.tsx#L13

        > error TS2353: Object literal may only specify known properties, and ''--pf-v5-c-label__content--Color'' does not exist in type 'Properties<string | number, string & {}>'.

        > The expected type comes from property 'style' which is declared here on type 'IntrinsicAttributes & LabelProps'

        Solution: Add `as CSSProperties` type assertion.

3. `yarn start` in ui

### CODE REVIEW

File: ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationContent.tsx [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
 import TechPreviewLabel from 'Components/PatternFly/TechPreviewLabel';
 
 type NavigationContentVariant = 'Deprecated' | 'TechPreview';
 
+const style = {
+    '--pf-v5-c-label__content--Color': 'var(--pf-v5-global--Color--dark-100)',
+} as CSSProperties;
+// Type assertion prevents TypeScript error:
+// error TS2353: Object literal may only specify known properties, and ''--pf-v5-c-label__content--Color'' does not exist in type 'Properties<string | number, string & {}>'.
+// The expected type comes from property 'style' which is declared here on type 'IntrinsicAttributes & LabelProps'
```
Comment (dvail) [inline ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationContent.tsx:13]: Is it possible that something in the React types caused our [override here](https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/react-app-env.d.ts#L16) to fail? I wonder if there is an alternative we can use that still allows us to omit the type assertions when setting PatternFly CSS vars?

Comment (pedrottimark) [reply ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationContent.tsx:13]: Aha, good point. Looks like only 3 commits after 18.3

https://github.com/DefinitelyTyped/DefinitelyTyped/commits/master/types/react

Comment (pedrottimark) [reply ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationContent.tsx:13]: Okay, it looks like the following contribution opened the door:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/67583

And the following revert closed the door:
https://github.com/DefinitelyTyped/DefinitelyTyped/commit/79bae95b770c7209689eacc033d93ce9998f4816

Comment (dvail) [reply ui/apps/platform/src/Containers/MainPage/Sidebar/NavigationContent.tsx:13]: It was fun while it lasted 😭 

I can re-investigate at another point in time to see if we can still get this ability, but it isn't a high priority right now.

**Review by dvail — APPROVED** [review]
Thanks for keeping on top of these, especially with the recent requests to update dev dependencies in older versions.

**Pull request merged**